uui-core/src/helpers/orderBy.ts

```diff
@@ -0,0 +1,17 @@
+import { compareScalars, getOrderComparer } from '../data/querying/getOrderComparer';
+
+export function orderBy<T>(arr: T[], by: ((item: T) => string | number | null) | string, order: 'asc' | 'desc' = 'asc') {
+    const sortedArr = [...arr];
+
+    if (typeof by === 'function') {
+        const sign = order === 'desc' ? -1 : 1;
+        return sortedArr.sort((a, b) => {
+            const aBy = by(a);
+            const bBy = by(b);
+            return compareScalars(aBy, bBy, sign);
+        });
+    }
+
+    const comparer = getOrderComparer([{ field: by, direction: order }]);
```

Comment:
We use a simple string comparator if `by` is a function. However, we use Intl-based comparator (implied by use of getOrderComparer) - if we pass a field name as a string.

Why do we need an overloads here? Probably we can accept a function only, and rewrite any `orderBy(arr, 'name')` to orderBy(arr, i => i.name)?